docs/user/rstudio/ide/guide/code/debugging.qmd

```diff
@@ -60,25 +63,25 @@ biggest_palindrome()
 
 ![](images/breakpoint-environment.png){fig-alt="A screenshot of the RStudio Environment pane, indicating a breakpoint function with a red circle." width="250"}
 
-If the function object doesn't exist yet (for instance, because you haven't called `source()` on the file), or the function object doesn't match the contents of the editor (for instance, because you've changed the file since the last `source()`), the breakpoint will be deferred.
+If the function object doesn't exist yet (for instance, because `source()` hasn't been called on the file), or the function object doesn't match the contents of the editor (for instance, because the file has changed since the last `source()`), the breakpoint will be deferred.
 
 ![](images/breakpoint-sourced.png){fig-alt="A screenshot of example R code and a warning that 'breakpoints will be activated when this file is sourced.'" width="350"}
 
-The circle outline indicates that RStudio is aware of the breakpoint, but that it hasn't been injected. In most cases, you can `source()` the file to resolve the problem.
+The hollow red circle indicates that RStudio is aware of the breakpoint, but that it hasn't been injected. In most cases, call `source()` on the file to resolve this problem.
 
 #### browser() breakpoints
 
-The R function [`browser()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/browser.html) halts execution and invokes an environment browser when it is called. You can put `browser()` anywhere in your code to stop at that point in the code for debugging. Here, for instance, it's used to halt when a function is about to return `TRUE`:
+The R function [`browser()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/browser.html) halts execution and invokes an environment browser when it is called. A `browser()` call can be added anywhere in the code to stop at that point in the code for debugging. Here, for instance, it's used to halt when a function is about to return `TRUE`:
 
 ![](images/browser-debug.png){fig-alt="A screenshot of R code with a browser() call, along with the traceback." width="450"}
 
-Unlike an editor breakpoint, the `browser()` statement is actually part of your code, so it needs to be applied like any other code change in order to become active (by sourcing the containing document, rebuilding the containing package, reloading the Shiny application, etc.).
+Unlike an editor breakpoint, the `browser()` statement is actually part of the code, so it needs to be applied like any other code change in order to become active (by sourcing the containing document, rebuilding the containing package, reloading the Shiny application, etc.).
 
 The environment browser invoked by `browser()` is the same one used in all of the other debugging facilities, so it can be considered the lowest-level debugging tool. Because the `browser()` statement requires no special tooling, it can be used in contexts where editor breakpoints won't work.
 
```

Comment:
What are some examples where the editor breakpoints won't work? I'm trying to create a distinction between the editor and browser tools and having a couple more examples would be good. The example below showing conditional debugging with browser is excellent.

Reply:
This is a good point, I'll think on it and leaving open for now.

Reply:
I think I've settled on linking out to Advanced R - we want to keep the RStudio guide more minimally focused on the IDE features and external resources like the Advanced R debugging chapter (which is linked twice) for strategy.